app/src/main/java/com/nextcloud/talk/activities/MainActivity.kt

```diff
@@ -152,7 +127,24 @@ class MainActivity : BaseActivity(), ActionBarProvider {
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
 
-        if (intent.hasExtra(BundleKeys.KEY_FROM_NOTIFICATION_START_CALL)) {
+        if (intent.action == BundleKeys.KEY_NEW_CONVERSATION) {
+            openNewConversationScreen()
+        } else if (intent.action == BundleKeys.KEY_OPEN_CONVERSATION) {
+            GlobalScope.launch {
+                val user: UserNgEntity? = usersRepository.getUserWithId(intent.getLongExtra(BundleKeys.KEY_INTERNAL_USER_ID, -1))
+                user?.let {
```

Comment:
I'd been running into the restrictions of `persistable bundle` and was thinking to only put the user id into the intent, but I'd have never figured this out, thanks! 😂

Reply:
:) I'd still argue we shouldn't be putting the whole user entity into the bundle, but for now it'll do as I do this all across the codebase anyway xD